addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/handler/YeelightHandlerBase.java

```diff
@@ -0,0 +1,226 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.yeelight.handler;
+
+import static org.openhab.binding.yeelight.YeelightBindingConstants.*;
+
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.HSBType;
+import org.eclipse.smarthome.core.library.types.IncreaseDecreaseType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.yeelight.YeelightBindingConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.yeelight.sdk.device.ConnectState;
+import com.yeelight.sdk.device.DeviceBase;
+import com.yeelight.sdk.device.DeviceFactory;
+import com.yeelight.sdk.device.DeviceStatus;
+import com.yeelight.sdk.enums.DeviceAction;
+import com.yeelight.sdk.enums.DeviceMode;
+import com.yeelight.sdk.enums.DeviceType;
+import com.yeelight.sdk.listeners.DeviceConnectionStateListener;
+import com.yeelight.sdk.listeners.DeviceStatusChangeListener;
+import com.yeelight.sdk.services.DeviceManager;
+
+/**
+ * The {@link YeelightHandlerBase} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Coaster Li - Initial contribution
+ */
+public abstract class YeelightHandlerBase extends BaseThingHandler
+        implements DeviceConnectionStateListener, DeviceStatusChangeListener {
+
+    private final Logger logger = LoggerFactory.getLogger(YeelightHandlerBase.class);
+    protected DeviceBase mDevice;
+
+    public YeelightHandlerBase(Thing thing) {
+        super(thing);
+    }
+
+    protected void updateUI(DeviceStatus status) {
+        thing.setLabel(status.getName().isEmpty() ? DeviceManager.getDefaultName(mDevice) : status.getName());
+    }
+
+    @Override
+    public void initialize() {
+        mDevice = DeviceFactory.build(getDeviceModel(getThing().getThingTypeUID()).name(), getThing().getUID().getId());
+        mDevice.setDeviceName(getThing().getLabel());
+        mDevice.setAutoConnect(true);
+        DeviceManager.getInstance().addDevice(mDevice);
+        mDevice.registerConnectStateListener(this);
+        mDevice.registerStatusChangedListener(this);
+        updateStatus(ThingStatus.OFFLINE);
```

Comment:
why OFFLINE? Better use UNKNOWN

Reply:
For reference: https://github.com/openhab/openhab2-addons/pull/3255#discussion_r178419421